### PR TITLE
Fix nuclear norm with requires_grad=True

### PR DESCRIPTION
**Pull request by vishwakftw**
Changelog:
- Selectively assign compute_uv in the at::svd used internally in the implementation of at::nuclear_norm

Test Plan:
- Add tests in common_method_invocations.py

Refixes: #18275

### CODE REVIEW

File: aten/src/ATen/native/LinearAlgebra.cpp [hunk 1]
```diff
@@ -546,29 +546,29 @@ Tensor nuclear_norm(const Tensor& self, bool keepdim) {
...
-  return at::sum(std::get<1>(at::svd(p, /*some=*/true, /*compute_uv=*/false)), -1, keepdim);
+  return at::sum(std::get<1>(at::svd(p, /*some=*/true, /*compute_uv=*/self.requires_grad())), -1, keepdim);
 }
 
 Tensor& nuclear_norm_out(Tensor& result, const Tensor& self, IntArrayRef dim, bool keepdim) {
   TORCH_CHECK(dim.size() == 2, "nuclear norm requires a 'dim' argument of size 2");
 
   Tensor p = _move_to_end(self, dim);
-  return at::sum_out(result, std::get<1>(at::svd(p, /*some=*/true, /*compute_uv=*/false)), -1, keepdim);
+  return at::sum_out(result, std::get<1>(at::svd(p, /*some=*/true, /*compute_uv=*/self.requires_grad())), -1, keepdim);
```
Comment (SsnL) [inline aten/src/ATen/native/LinearAlgebra.cpp]: does this respect `GradMode` yet?

Comment (vishwakftw) [reply aten/src/ATen/native/LinearAlgebra.cpp]: I'm not sure I follow, are you asking if this fix works?

Comment (SsnL) [reply aten/src/ATen/native/LinearAlgebra.cpp]: Sorry for not being clear. I am asking if `.requires_grad` respects `GradMode` now (e.g., set by `no_grad()`). @yf225 probably knows..

Comment (vishwakftw) [reply aten/src/ATen/native/LinearAlgebra.cpp]: No worries, thanks for the clarification. I’m not sure about it either.

Comment (yf225) [reply aten/src/ATen/native/LinearAlgebra.cpp]: `.requires_grad` doesn't respect `GradMode` (and probably won't based on the concerns at https://github.com/pytorch/pytorch/pull/22506#pullrequestreview-259641452). Another concern is that `self` might not be a Variable when the user calls this function, and we should likely use `GradMode::is_enabled() && self.is_variable() && self.requires_grad()` here to be safe.

Comment (vishwakftw) [reply aten/src/ATen/native/LinearAlgebra.cpp]: Thank you @yf225, I’ll change as advised.

Comment (vishwakftw) [reply aten/src/ATen/native/LinearAlgebra.cpp]: @yf225 I've made the change, could you please review when you get the time?

**Review by yf225 — APPROVED** [review]
`self.requires_grad()` related change looks good to me. @SsnL Would you like to review the other parts of the PR?

**Review by ssnl — APPROVED** [review]
LGTM thank you!!!!!!

**Pull request closed without merging**